filebeat/input/filestream/prospector.go

```diff
@@ -71,18 +68,52 @@ func newFileProspector(
 	}, nil
 }
 
+func (p *fileProspector) Init(cleaner loginp.ProspectorCleaner) error {
+	if p.cleanRemoved {
+		cleaner.CleanIf(func(key string, u loginp.Unpackable) bool {
+			var st state
+			err := u.UnpackCursor(&st)
+			if err != nil {
+				// remove faulty entries
+				return true
+			}
+
+			_, err = os.Stat(st.Source)
+			if err != nil {
+				return true
+			}
+			return false
+		})
+	}
+
+	identifierName := p.identifier.Name()
+	cleaner.UpdateIdentifiers(func(key string, u loginp.Unpackable) (bool, string) {
+		var st state
+		err := u.UnpackCursor(&st)
```

Comment:
the prospector should not be allowed to read the cursor state. The cursor state is updated after events are ACKed. Better separate the `state` struct into actual state (`type state struct { offset int64 }` and meta-data like `type source struct { Path string; IdenfifierName string }`